pkg/cloudprovider/aws.go

```diff
@@ -71,20 +72,34 @@ func (a *AWS) AssignPrivateIP(ip net.IP, node *corev1.Node) error {
 	networkInterface := networkInterfaces[0]
 	addIP := ip.String()
 	keepIPs := []*string{}
+	newIPs := []*string{}
 	if utilnet.IsIPv6(ip) {
 		for _, assignedIPv6 := range networkInterface.Ipv6Addresses {
 			if assignedIP := net.ParseIP(*assignedIPv6.Ipv6Address); assignedIP != nil && assignedIP.Equal(ip) {
 				return AlreadyExistingIPError
 			}
 			keepIPs = append(keepIPs, assignedIPv6.Ipv6Address)
 		}
-		keepIPs = append(keepIPs, &addIP)
+		newIPs = append(newIPs, &addIP)
 		input := ec2.AssignIpv6AddressesInput{
 			NetworkInterfaceId: networkInterface.NetworkInterfaceId,
-			Ipv6Addresses:      keepIPs,
+			Ipv6Addresses:      newIPs,
 		}
 		_, err = a.client.AssignIpv6Addresses(&input)
 		if err != nil {
+			var newIPsStr string
+			var keepIPsStr string
+			for _, ip := range newIPs {
+				newIPsStr = fmt.Sprintf("%s %s", newIPsStr, *ip)
+			}
+			for _, ip := range keepIPs {
+				keepIPsStr = fmt.Sprintf("%s %s", keepIPsStr, *ip)
+			}
+			klog.Warningf("error: %s, tried to assign the following %d IP addresses to the interface:%v. Already got %d IP addresses on the interface: %v",
+				err.Error(),
+				len(newIPs), newIPsStr,
+				len(keepIPs), keepIPsStr,
+			)
 			return err
 		}
 		return a.waitForCompletion(node, awsapi.StringValueSlice(keepIPs), false)
```

Comment:
I think this should change to `return a.waitForCompletion(node, awsapi.StringValueSlice(newIPs), false)`